### PR TITLE
tests: make `TestExternalTS` stable

### LINKED ISSUE #5682: `TestExternalTimestamp` failed
### Which jobs are failing
```
--- FAIL: TestExternalTimestamp (2.22s)
    cluster_test.go:1529: 
        	Error Trace:	cluster_test.go:1529
        	Error:      	Not equal: 
        	            	expected: 0xe9
        	            	actual  : 0x611088156b00002
        	Test:       	TestExternalTimestamp
```

### CI link
https://github.com/tikv/pd/actions/runs/3392914559/jobs/5639655252

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Close #5682

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/cluster/cluster_test.go [hunk 1]
```diff
@@ -1503,32 +1503,58 @@ func TestExternalTimestamp(t *testing.T) {
...
-		re.NoError(err)
-		globalTS := tsoutil.GenerateTS(resp.Timestamp)
-
-		req2 := &pdpb.SetExternalTimestampRequest{
-			Header:    testutil.NewRequestHeader(clusterID),
-			Timestamp: globalTS + 100,
-		}
-		_, err = grpcPDClient.SetExternalTimestamp(context.Background(), req2)
-		re.NoError(err)
+		testutil.Eventually(re, func() bool {
```
Comment (rleungx) [inline tests/server/cluster/cluster_test.go:1509]: Why use Eventually?

Comment (lhy1024) [reply tests/server/cluster/cluster_test.go:1509]: L1528 tries to set am external ts large than global ts in L1526. We need to ensure the global ts of L1528 still be less than L1528.

Comment (lhy1024) [reply tests/server/cluster/cluster_test.go:1509]: Maybe 10s is enough.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**